.github/workflows/main.yml

```diff
@@ -0,0 +1,45 @@
+name: CI
+
+on: 
+  # any time you create a pull request against `main` or 
+  # try to push to `main`, this workflow file will run 
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+# the jobs are the workflows that github will automatically run. 
+# the steps that need to be taken for this workflow to pass are:
+# checkout the branch
+# useNode.js
+# install dependencies
+# build the thing
+# run the linting scripts
+# run the cypress tests
+# run the jest tests
+ 
+jobs:
+  build-lint-and-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [18.x]
```

Comment:
To be perfectly honest, I nabbed this from the accessible-components repo. I don't quite understand why it's here. I think it means that during the `build-lint-and-test` job, github will use Nodev18. But, do I even need that here? I don't quite the distinction yet.

Reply:
A matrix allows you to run a job for multiple configurations using different dependencies. So for example, if you were creating an npm package, you might want to specify multiple versions of node to test against. The job will run for each of those versions.

 In your case, it's fine to have a single matrix that matches the node version you use for local development. I'd recommend adding a corresponding `engines` entry [to your package.json](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines) to ensure consistency between your local environment and CI.

I also think you need to add another setting to ensure that you're using your specified version of node. Your [workflow run](https://github.com/marissahuysentruyt/marissahuysentruyt/actions/runs/6746947969/job/18342002323#step:3:8) shows that it is using an 18.x version of node, but I think this may simply be using the default. To ensure that `setup-node` is using the specified matrix version, you should add `with` and `node-version` settings to that step. Scroll down to the "Basic" usage example [here](https://github.com/actions/setup-node#usage) to see an example:

```yml
- uses: actions/setup-node@v3
  with:
    node-version: 18
```